Application/EdFi.Ods.Common/Providers/Queries/Criteria/IdentificationCodeAggregateQueryCriteriaApplicator.cs

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using EdFi.Common.Configuration;
+using EdFi.Ods.Common.Database.Querying;
+using EdFi.Ods.Common.Descriptors;
+using EdFi.Ods.Common.Models.Domain;
+using EdFi.Ods.Common.Providers.Criteria;
+
+namespace EdFi.Ods.Common.Providers.Queries.Criteria;
+
+/// <summary>
+/// Applies criteria to a query based on identification codes property values
+/// </summary>
+public class IdentificationCodeAggregateQueryCriteriaApplicator : IAggregateRootQueryCriteriaApplicator
+{
+    private readonly IDescriptorResolver _descriptorResolver;
+    private readonly DatabaseEngine _databaseEngine;
+    private const string IdentificationCodeTableAlias = "idct";
+
+    public IdentificationCodeAggregateQueryCriteriaApplicator(
+        IDescriptorResolver descriptorResolver,
+        DatabaseEngine databaseEngine)
+    {
+        _descriptorResolver = descriptorResolver;
+        _databaseEngine = databaseEngine;
+    }
+
+    public void ApplyAdditionalParameters(QueryBuilder queryBuilder, Entity entity, AggregateRootWithCompositeKey specification,
+        IDictionary<string, string> additionalParameters)
+    {
+        if (additionalParameters == null || !additionalParameters.Any() || additionalParameters.All(
```

Comment:
I think you could leave the `null` and the `Any` checks, but I don't think the latter is needed because ASP.NET will fill that dictionary with values that were not bindable to the other models in the controller signature. We should confirm this, but I think this will be wasted worked. Thoughts/concerns?